tvbextxircuits/nb_generator.py

```diff
@@ -78,14 +78,51 @@ def _remove_file(self, filename):
             os.rmdir(self.model_configs_folder)
 
 
+class NotebookFactory(object):
+
+    @staticmethod
+    def get_notebook_for_component(component_name, component_id, component_path, component_inputs):
+        component_class = determine_component_class(component_name, component_path)
+
+        if not issubclass(component_class, ComponentWithWidget):
+            return None
+
+        if component_class.__name__ == 'StoreResults':
```

Comment:
Maybe in the future this factory mapping to `*NotebookGenerators` could be improved (like keeping a dict somehwere). But with very few `*NotebookGenerators` as right now, the code looks very good as it is.